src/pops.cpp

```diff
@@ -45,269 +44,226 @@ bool all_infected(IntegerMatrix susceptible)
+constexpr typename build_indices<std::tuple_size<Bare<Tuple>>::value>::type
+make_indices() {
+  return {};
+}
+
+template <typename Tuple, int... Indices>
+std::array<typename std::tuple_element<0, Bare<Tuple>>::type,
+           std::tuple_size<Bare<Tuple>>::value>
+to_array(Tuple &&tuple, indices<Indices...>) {
   using std::get;
-  return {{ get<Indices>(std::forward<Tuple>(tuple))... }};
+  return {{get<Indices>(std::forward<Tuple>(tuple))...}};
 }
 
-template<typename Tuple>
-auto to_array(Tuple&& tuple)
-  -> decltype( to_array(std::declval<Tuple>(), make_indices<Tuple>()) )
-  {
-    return to_array(std::forward<Tuple>(tuple), make_indices<Tuple>());
-  }
+template <typename Tuple>
+auto to_array(Tuple &&tuple)
+    -> decltype(to_array(std::declval<Tuple>(), make_indices<Tuple>())) {
+  return to_array(std::forward<Tuple>(tuple), make_indices<Tuple>());
+}
 
 // [[Rcpp::interfaces(r, cpp)]]
 // [[Rcpp::plugins(cpp11)]]
 // [[Rcpp::export]]
-List pops_model(int random_seed, 
-                bool use_lethal_temperature, double lethal_temperature, int lethal_temperature_month,
-                IntegerMatrix infected,
-                std::vector<IntegerMatrix> exposed,
-                IntegerMatrix susceptible,
-                IntegerMatrix total_plants,
-                bool mortality_on,
-                IntegerMatrix mortality_tracker,
-                IntegerMatrix mortality,
-                std::vector<NumericMatrix> treatment_maps,
-                std::vector<std::string> treatment_dates,
-                std::vector<int> pesticide_duration,
-                IntegerMatrix resistant,
-                bool use_movements, std::vector<std::vector<int>> movements,
-                std::vector<std::string> movements_dates,
-                bool weather,
-                std::vector<NumericMatrix> temperature,
-                std::vector<NumericMatrix> weather_coefficient,
-                double ew_res, double ns_res, int num_rows, int num_cols,
-                std::string time_step, double reproductive_rate,
-                double mortality_rate = 0.0, int mortality_time_lag = 2,
-                int season_month_start = 1, int season_month_end = 12,
-                std::string start_date = "2018-01-01", std::string end_date = "2018-12-31",
-                std::string treatment_method = "ratio",
-                std::string natural_kernel_type = "cauchy", 
-                std::string anthropogenic_kernel_type = "cauchy", 
-                bool use_anthropogenic_kernel = false, double percent_natural_dispersal = 0.0,
-                double natural_distance_scale = 21, double anthropogenic_distance_scale = 0.0, 
-                std::string natural_dir = "NONE", double natural_kappa = 0,
-                std::string anthropogenic_dir = "NONE", double anthropogenic_kappa = 0,
-                std::string output_frequency = "year", std::string model_type_ = "SI",
-                int latency_period = 0
-)
-{
-  
+List pops_model(
```

Comment:
I would like to move function calls to have a single variable on each line. I started doing this with the R functions but haven't done that here yet. It made it much easier to view changes.